.github/workflows/build.yml

```diff
@@ -6,22 +6,32 @@ on:
   workflow_dispatch:
 
 jobs:
-  check-formatting:
-    name: Check source code formatting
+  lint:
+    name: Lint
     runs-on: ubuntu-latest
     steps:
       - name: Checkout sources
         uses: actions/checkout@v3
 
-      - name: Check source code formatting
+      - name: Running source code format check
         uses: jidicula/clang-format-action@v4.9.0
         with:
           clang-format-version: "15"
           check-path: "src"
 
+      - name: Running static analysis
+        uses: cpp-linter/cpp-linter-action@v2
+        id: linter
+        with:
+          style: file
+          tidy-checks: ""
```

Comment:
Probably good to specify latest version (`14`) instead of default (`12`):
```
version: 14
```

Reply:
Good point! I'll have a few more test runs and check the results. 
Currently, the action does not seem to work as I thought it would. For example, in 
https://github.com/engelmi/orch/actions/runs/3641242250/workflow#L27
the test directory is not ignored and clang-tidy should throw `magic number errors` for [these lines](https://github.com/engelmi/orch/blob/main/test/common/parse-util_test.c#L33-L38), but doesn't. 
If it doesn't work, I'd switch to the make targets.